lib/rules/declaration-property-max-values/index.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const matchesStringOrRegExp = require('../../utils/matchesStringOrRegExp');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const valueParser = require('postcss-value-parser');
+const vendor = require('../../utils/vendor');
+const validateOptions = require('../../utils/validateOptions');
+const { isNumber } = require('../../utils/validateTypes');
+const validateObjectWithProps = require('../../utils/validateObjectWithProps');
+
+const ruleName = 'declaration-property-max-values';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (property, max) =>
+		`Expected "${property}" to have no more than ${max} ${max === 1 ? 'value' : 'values'}`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/list/declaration-property-max-values',
+};
+
+const isValueNode = (/** @type {valueParser.Node} */ node) => {
+	return node.type === 'word' || node.type === 'function' || node.type === 'string';
+};
+
+/** @type {import('stylelint').Rule<Record<string, number>>} */
+const rule = (primary) => {
+	return (root, result) => {
+		root.walkDecls((decl) => {
+			const validOptions = validateOptions(result, ruleName, {
+				actual: primary,
+				possible: [validateObjectWithProps(isNumber)],
+			});
+
+			if (!validOptions) {
+				return;
+			}
+
+			const { prop, value } = decl;
+			const propLength = valueParser(value).nodes.filter(isValueNode).length;
+
+			const unprefixedProp = vendor.unprefixed(prop);
+			const propKey = Object.keys(primary).find((propIdentifier) =>
+				matchesStringOrRegExp(unprefixedProp, propIdentifier),
+			);
+
+			if (!propKey) {
+				return;
+			}
+
+			const max = primary[propKey];
+
+			if (propLength <= max) {
+				return;
+			}
+
+			report({
+				message: messages.rejected(prop, max),
+				node: decl,
```

Comment:
[ask] Which column position should we specify: property or value?

```suggestion
				node: decl,
				index: declarationValueIndex(decl),
```

property:

<img width="410" alt="image" src="https://user-images.githubusercontent.com/473530/155244718-efc6b5e3-327c-4ba6-ae81-90e63da3a9b9.png">

value:

<img width="474" alt="image" src="https://user-images.githubusercontent.com/473530/155244750-d9fd483a-3b60-4b9c-b45a-3be74054a622.png">

Reply:
I am not sure, though I'm inclined to say property (since to me, this rule is about properties meeting a certain predicate) - what do you think?

Reply:
Indeed, this rule aims at a *declaration-property*, not a *declaration property value*. So let's keep it as-is. Thanks for the answer. 👍🏼